lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -363,6 +358,18 @@ LogicalResult CircuitOp::verify() {
             << "previous extmodule definition occurred here";
         return failure();
       }
+      if (!aType.isa<FIRRTLBaseType>() || !bType.isa<FIRRTLBaseType>())
+        return extModule.emitOpError().append(
```

Comment:
Note: Maybe in the future we can add support for `RefType` ports on ext Modules, for an XMR into an extern module.

Reply:
I'll add logic to make this possible in this verifier, thanks!

Reply:
Actually, this code is checking types are compatible (?) for colliding defnames on extmodules (in the presence of parameters), and I'm not sure it makes sense to do the same for RefType's-- those aren't as flexible re:automatically extending/truncating widths, so comparing them as widthless doesn't make sense as if they don't match they'll fail verification at point of use I think?

How about we allow RefTypes at this point, but they must match?

Reply:
Thanks, this makes sense.